src/command_modules/azure-cli-datalake/azure/cli/command_modules/datalake/analytics/custom.py

```diff
@@ -0,0 +1,229 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import time
+import uuid
+
+from azure.cli.core.prompting import prompt_pass, NoTTYException
+from azure.mgmt.datalake.analytics.account.models import (DataLakeAnalyticsAccountUpdateParameters,
+                                                          FirewallRule,
+                                                          DataLakeAnalyticsAccount,
+                                                          DataLakeStoreAccountInfo)
+
+from azure.mgmt.datalake.analytics.job.models import (JobType,
+                                                      JobState,
+                                                      JobInformation,
+                                                      USqlJobProperties)
+# pylint: disable=line-too-long
+from azure.mgmt.datalake.analytics.catalog.models import (DataLakeAnalyticsCatalogCredentialCreateParameters,
+                                                          DataLakeAnalyticsCatalogCredentialUpdateParameters)
+from azure.cli.core.commands.client_factory import get_mgmt_service_client
+from azure.cli.core._util import CLIError
+import azure.cli.core.azlogging as azlogging
+
+logger = azlogging.get_az_logger(__name__)
+# account customiaztions
+def list_adla_account(client, resource_group_name=None):
+    account_list = client.list_by_resource_group(resource_group_name=resource_group_name) \
+        if resource_group_name else client.list()
+    return list(account_list)
+
+# pylint: disable=too-many-arguments
+def create_adla_account(client,
+                        resource_group_name,
+                        account_name,
+                        default_datalake_store,
+                        location=None,
+                        tags=None,
+                        max_degree_of_parallelism=30,
+                        max_job_count=3,
+                        query_store_retention=30,
+                        tier=None):
+    adls_list = list()
+    adls_list.append(DataLakeStoreAccountInfo(default_datalake_store))
+    location = location or _get_resource_group_location(resource_group_name)
+    create_params = DataLakeAnalyticsAccount(location,
+                                             default_datalake_store,
+                                             adls_list)
+    if tags:
+        create_params.tags = tags
+
+    if max_degree_of_parallelism:
+        create_params.max_degree_of_parallelism = max_degree_of_parallelism
+
+    if max_job_count:
+        create_params.max_job_count = max_job_count
+
+    if query_store_retention:
+        create_params.query_store_retention = query_store_retention
+
+    if tier:
+        create_params.new_tier = tier
```

Comment:
You don't need these if statements. Just pass all of these values in the DataLakeAnalyticsAccount constructor, the way you did below in your update method.